pingcap/tidb
# add binlog socket config for compatible with kafka version pump

### What problem does this PR solve? 
add binlog socket config for compatible with kafka version pump

### What is changed and how it works?
add a new config binlogsocket
if binlogsocket is not empty, will use `pumpcli.NewLocalPumpsClient` to create a pump client, and this pump client will only write binlog to the local pump by this socket file.

### Check List 

Tests 

 - Integration test
 - Manual test
   set binlogsocket, and deploy kafka version pump, write data and then check


Related changes

 - Need to cherry-pick to the release branch
 - Need to update the documentation
 - Need to update the `tidb-ansible` repository
 - Need to be included in the release note
